tests/tests/it/old_shares.rs

```diff
@@ -0,0 +1,45 @@
+static OLD_SHARES: include_dir::Dir =
+    include_dir::include_dir!("$CARGO_MANIFEST_DIR/../test-data/old-shares");
+
+#[test]
+fn deserialize_old_shares() {
+    for share in OLD_SHARES.files() {
+        let file_name = share
+            .path()
+            .file_name()
+            .expect("share doesn't have file name")
+            .to_str()
+            .expect("share name is not valid utf8");
+
+        if file_name == "README.md" {
+            continue;
+        } else if file_name.contains("secp256k1") {
+            deserialize_old_share_on_curve::<cggmp21::supported_curves::Secp256k1>(share);
+        } else if file_name.contains("secp256r1") {
+            deserialize_old_share_on_curve::<cggmp21::supported_curves::Secp256r1>(share);
+        } else if file_name.contains("stark") {
+            deserialize_old_share_on_curve::<cggmp21::supported_curves::Stark>(share);
+        } else {
+            panic!("couldn't figure out the curve from the share name {file_name}")
+        }
+    }
+}
+
+fn deserialize_old_share_on_curve<E: generic_ec::Curve>(share: &include_dir::File) {
+    let ext = share
+        .path()
+        .extension()
+        .expect("share file name doesn't have extension")
+        .to_str()
+        .expect("key share ext is not valid utf8");
+
+    let _: cggmp21::IncompleteKeyShare<E> = match ext {
```

Comment:
Might add checking if threshold and hd properties hold after deserialization. Though I think there's a low chance this goes wrong, so might be overkill. (I thought that was the reason you structured file names this way)

Reply:
I just wanted to have every possible "kind" of key share. E.g. whether key share has or doesn't a chain code may result into a different kind of key share

Reply:
For validation - I assume that `Valid<_>` is good enough alone to validate a share

Reply:
I meant that, e.g., if the name has "hd-true", we verify that chaincode is present, and if "hd-false", verify that it's absent